Services/Certificate/classes/API/Repository/UserDataRepository.php

```diff
@@ -129,41 +121,97 @@ public function getUserData(UserDataFilter $filter, array $ilCtrlStack) : array
         return $result;
     }
 
+
+    /**
+     * @param UserDataFilter $filter
+     *
+     * @return int
+     */
+    public function getUserCertificateDataMaxCount(UserDataFilter $filter) : int
+    {
+        $sql = 'SELECT COUNT(il_cert_user_cert.id) as count
+        ' . $this->getQuery($filter,true);
+
+        $result = $this->database->query($sql);
+
+        $max_count = intval($this->database->fetchAssoc($result)["count"]);
+
+        return $max_count;
+    }
+
+
+    /**
+     * @param UserDataFilter $filter
+     * @param bool           $max_count_only
+     *
+     * @return string
+     */
+    private function getQuery(UserDataFilter $filter, bool $max_count_only = false) : string
+    {
+        $userIds = $filter->getUserIds();
+
+        $sql = 'FROM il_cert_user_cert
+LEFT JOIN object_data ON object_data.obj_id = il_cert_user_cert.obj_id
+LEFT JOIN object_data_del ON object_data_del.obj_id = il_cert_user_cert.obj_id
+LEFT JOIN object_reference ON object_reference.obj_id = il_cert_user_cert.obj_id
+INNER JOIN usr_data ON usr_data.usr_id = il_cert_user_cert.user_id
+WHERE ' . $this->database->in('il_cert_user_cert.user_id', $userIds, false, 'integer')
+            . $this->createWhereCondition($filter);
+
+        if (!$max_count_only) {
+            $sql.=  ' ' . $this->createOrderByClause($filter);
+            $this->database->setLimit($filter->getLimitEnd(), $filter->getLimitStart());
+        }
+
+        return $sql;
+    }
+
+
     /**
      * @param UserDataFilter $filter
      * @return string
      */
     private function createOrderByClause(UserDataFilter $filter) : string
     {
-        $orderBy = ' ORDER BY ';
+        $sorts = $filter->getSorts();
 
-        switch (true) {
-            case $filter->shouldSortByLogins():
-                $orderBy .= 'usr_data.login';
-                break;
+        if (!empty($sorts)) {
 
-            case $filter->shouldSortByFirstNames():
-                $orderBy .= 'usr_data.firstname';
-                break;
+            $orderBy = ' ORDER BY ';
 
-            case $filter->shouldSortByLastNames():
-                $orderBy .= 'usr_data.lastname';
-                break;
+            foreach ($sorts as [$key, $reverse]) {
+                $reverse = $reverse ? ' DESC ' : ' ASC ';
```

Comment:
Please rename this variable to `$direction` or something similar.

Reply:
Made in https://github.com/ILIAS-eLearning/ILIAS/pull/2216